code/modules/pool/pool_main.dm

```diff
@@ -0,0 +1,278 @@
+/turf/open/pool
+	icon = 'icons/turf/pool.dmi'
+	name = "poolwater"
+	desc = "You're safer here than in the deep."
+	icon_state = "pool_tile"
+	heat_capacity = INFINITY
+	var/filled = TRUE
+	var/next_splash = 1
+	var/obj/effect/overlay/water/watereffect
+	var/obj/effect/overlay/water/top/watertop
+	var/obj/machinery/pool/controller/controller
+
+
+/turf/open/pool/Initialize()
+	watereffect = new /obj/effect/overlay/water(src)
+	watertop = new /obj/effect/overlay/water/top(src)
+	. = ..()
+
+/turf/open/pool/Destroy()
+	QDEL_NULL(watereffect)
+	QDEL_NULL(watertop)
+	controller = null
+	return ..()
+
+/turf/open/pool/proc/update_icon()
+	if(!filled)
+		name = "drained pool"
+		desc = "No diving!"
+		QDEL_NULL(watereffect)
+		QDEL_NULL(watertop)
+	else
+		name = "poolwater"
+		desc = "You're safer here than in the deep."
+		watereffect = new /obj/effect/overlay/water(src)
+		watertop = new /obj/effect/overlay/water/top(src)
+
+/obj/effect/overlay/water
+	name = "water"
+	icon = 'icons/turf/pool.dmi'
+	icon_state = "bottom"
+	density = 0
+	mouse_opacity = 0
+	layer = ABOVE_MOB_LAYER
+	anchored = TRUE
+
+/obj/effect/overlay/water/top
+	icon_state = "top"
+	layer = BELOW_MOB_LAYER
+
+/mob/living
+	var/swimming = FALSE
```

Comment:
What the hell is THIS

Reply:
this should be a movement type imo

Reply:
it's for a bunch of shitcode checks, turning it into a movement type would be a good idea though.